### ISSUE TITLE
Sundials gives an error under Windows (invalid subtyping in definition of AbstractSundialsIntegrator)

### ISSUE THREAD

**Issue by MartinOtter** [JuliaDiffEq/Sundials.jl#196]
Sundials gives an error under Windows:
```
]add Sundials
test Sundials

ERROR: LoadError: LoadError: invalid subtyping in definition of AbstractSundialsIntegrator
Stacktrace:
 [1] top-level scope at none:0
 [2] include at .\boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at .\loading.jl:1044
 [4] include at .\sysimg.jl:29 [inlined]
 [5] include(::String) at HOME\.julia\packages\Sundials\AL2Km\src\Sundials.jl:3
 [6] top-level scope at none:0
 [7] include at .\boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at .\loading.jl:1044
 [9] include(::Module, ::String) at .\sysimg.jl:29
 [10] top-level scope at none:2
 [11] eval at .\boot.jl:319 [inlined]
 [12] eval(::Expr) at .\client.jl:393
 [13] top-level scope at .\none:3
in expression starting at HOME\.julia\packages\Sundials\AL2Km\src\common_interface\integrator_types.jl:16
in expression starting at HOME\.julia\packages\Sundials\AL2Km\src\Sundials.jl:65
ERROR: LoadError: Failed to precompile Sundials [c3572dad-4567-51f8-b174-8c6c989267f4] to HOME\.julia\compiled\v1.0\Sundials\j8Ppj.ji
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] macro expansion at .\logging.jl:311 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at .\loading.jl:1190
 [4] macro expansion at .\logging.jl:309 [inlined]
 [5] _require(::Base.PkgId) at .\loading.jl:947
 [6] require(::Base.PkgId) at .\loading.jl:858
 [7] macro expansion at .\logging.jl:309 [inlined]
 [8] require(::Module, ::Symbol) at .\loading.jl:840
 [9] include at .\boot.jl:317 [inlined]
 [10] include_relative(::Module, ::String) at .\loading.jl:1044
 [11] include(::Module, ::String) at .\sysimg.jl:29
 [12] include(::String) at .\client.jl:392
 [13] top-level scope at none:0
in expression starting at HOME\.julia\packages\Sundials\AL2Km\test\runtests.jl:1
ERROR: Package Sundials errored during testing
```

**Comment by ChrisRackauckas** [comment]
What version of Sundials and DiffEqBase are you using? You are mixing a new DiffEqBase with an old Sundials.

**Comment by MartinOtter** [comment]
Today, I made "add DifferentialEquations" and afterwards ModiaMath (that uses Sundials) gave an error when importing it. I then removed DifferentialEquations and added Sundials to try to figure out the reason. One experiment was "]add Sundials" leading to the compilation error above. Status shows `Sundials v2.6.0`.

When I remove Sundials and make "]activate ModiaMath" and then "using Sundials", then this works (so I guess the Sundials version is used that is defined in the project.toml file of ModiaMath). However, again `Sundials v2.6.0` is shown. So, I am confused.

When starting Julia 1.0.2 freshly, and giving the commands `]status`:
```
(v1.0) pkg> status
    Status `D:\otter\home\.julia\environments\v1.0\Project.toml`
  [336ed68f] CSV v0.4.3
  [944b1d66] CodecZlib v0.5.1
  [a93c6f00] DataFrames v0.15.1
  [864edb3b] DataStructures v0.14.0
  [e30172f5] Documenter v0.20.0
  [f6369f11] ForwardDiff v0.10.1
  [7073ff75] IJulia v1.15.2
  [682c06a0] JSON v0.20.0
  [cb905087] Modia v0.2.4-dev
  [9eeeb1f0] Modia3D v0.2.2-dev
  [67ccffd1] ModiaMath v0.4.0-dev
  [8d779e82] ModiaMedia v0.1.0
  [438e738f] PyCall v1.18.5
  [d330b81b] PyPlot v2.6.3
  [295af30f] Revise v1.0.2
  [90137ffa] StaticArrays v0.10.0
  [c3572dad] Sundials v2.6.0
  [1986cc42] Unitful v0.14.0
  [9e88b42a] Serialization
  [8dfed614] Test
```
and then `using Sundials`, the compilation error above occurs.

Note, the Continuous-Integration of Sundials on Appveyor shown at https://github.com/JuliaDiffEq/Sundials.jl gives an error ("failed 8 days ago").

I then removed DiffEqBase under `.Julia/compiled` and under  `.Julia/packages` and then `using Sundials`. This gave the error: `ERROR: LoadError: ArgumentError: Package DiffEqBase [2b5f629d-d688-5b77-993f-72d75c75574e] is required but does not seem to be installed: Run Pkg.instantiate() to install all recorded dependencies.`

O.k. called `]instantiate` leading to:
```
(v1.0) pkg> instantiate
  Updating registry at `HOME\.julia\registries\General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Installed DiffResults ─ v0.0.4
 Installed DiffRules ─── v0.0.9
 Installed DiffEqBase ── v5.2.3
```
`using Sundials` gives again the compilation error from above.

**Comment by MartinOtter** [comment]
I recognized that the current Sundials version is 3.0.0 and not v2.6.0. I made `]add Sundials@3.0.0` and then got the error:
```
(v1.0) pkg> add Sundials@3.0.0
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package DataStructures [864edb3b]:
 DataStructures [864edb3b] log:
 ├─possible versions are: [0.2.0, 0.2.2-0.2.15, 0.3.0-0.3.14, 0.4.0-0.4.6, 0.5.0-0.5.3, 0.6.0-0.6.1, 0.7.0-0.7.4, 0.8.1-0.8.4, 0.9.0, 0.10.0, 0.11.0-0.11.1, 0.12.0, 0.13.0, 0.14.0, 0.15.0] or uninstal
led
 ├─restricted to versions * by ModiaMath [67ccffd1], leaving only versions [0.2.0, 0.2.2-0.2.15, 0.3.0-0.3.14, 0.4.0-0.4.6, 0.5.0-0.5.3, 0.6.0-0.6.1, 0.7.0-0.7.4, 0.8.1-0.8.4, 0.9.0, 0.10.0, 0.11.0-0.
11.1, 0.12.0, 0.13.0, 0.14.0, 0.15.0]
 │ └─ModiaMath [67ccffd1] log:
 │   ├─possible versions are: 0.4.0 or uninstalled
 │   └─ModiaMath [67ccffd1] is fixed to version 0.4.0-dev
 ├─restricted to versions 0.14.0 by an explicit requirement, leaving only versions 0.14.0
 └─restricted by compatibility requirements with Sundials [c3572dad] to versions: 0.15.0 - no versions left
   └─Sundials [c3572dad] log:
     ├─possible versions are: [0.1.0-0.1.3, 0.2.0-0.2.2, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.9.0, 0.10.0, 0.11.0-0.11.1, 0.12.0, 0.13.0, 0.14.0-0.14.3, 0.15.0, 0.16.0, 0.17.0-0.17.1, 0.18.0-0.
18.3, 0.19.0, 1.0.0-1.0.1, 1.1.0-1.1.1, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.4, 1.6.0, 2.0.0, 2.1.0, 2.2.0, 2.3.0-2.3.1, 2.3.3, 2.4.0-2.4.1, 2.5.0-2.5.1, 2.6.0, 2.7.0, 3.0.0] or uninstalled
     ├─restricted to versions * by ModiaMath [67ccffd1], leaving only versions [0.1.0-0.1.3, 0.2.0-0.2.2, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.9.0, 0.10.0, 0.11.0-0.11.1, 0.12.0, 0.13.0, 0.14.0
-0.14.3, 0.15.0, 0.16.0, 0.17.0-0.17.1, 0.18.0-0.18.3, 0.19.0, 1.0.0-1.0.1, 1.1.0-1.1.1, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.4, 1.6.0, 2.0.0, 2.1.0, 2.2.0, 2.3.0-2.3.1, 2.3.3, 2.4.0-2.4.1, 2.5.0-2.5.1, 2.6
.0, 2.7.0, 3.0.0]
     │ └─ModiaMath [67ccffd1] log: see above
     └─restricted to versions 3.0.0 by an explicit requirement, leaving only versions 3.0.0
```

**Comment by MartinOtter** [comment]
O.k. used commands
```
activate ModiaMath
update

using ModiaMath
```
Now everything compiled and examples are running again (and Sundials v3.0.0 is used).

O.k. Problem is fixed.
